### PR TITLE
Detangle tea and tea-junit4

### PR DESCRIPTION
**Pull request by strooooke**

### CODE REVIEW

File: tea-junit4/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/build
```
Comment (maxhille) [inline tea-junit4/.gitignore:1]: newline missing

> Is it annoying?

Yes

> Is it necessary

Yes (https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline)

> How can I prevent this?
* use a proper editor for .gitignore
or
* IDEA: Settings -> General -> On Save -> "Ensure ..."

File: tea-junit4/build.gradle [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        jvmTarget = '1.8'
+    }
+}
+
+dependencies {
+    implementation project(':tea')
+    implementation 'junit:junit:4.13.2'
+    implementation 'androidx.test.espresso.idling:idling-concurrent:3.4.0'
+    implementation "androidx.compose.ui:ui-test-junit4:1.2.1"
+}
```
Comment (maxhille) [inline tea-junit4/build.gradle:71]: newline

File: tea-junit4/proguard-rules.pro [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+#   public *;
+#}
+
+# Uncomment this to preserve the line number information for
+# debugging stack traces.
+#-keepattributes SourceFile,LineNumberTable
+
+# If you keep the line number information, uncomment this to
+# hide the original source file name.
+#-renamesourcefileattribute SourceFile
```
Comment (maxhille) [inline tea-junit4/proguard-rules.pro:21]: newline

File: tea-junit4/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  ~ GNU General Public License for more details.
+  ~
+  ~ You should have received a copy of the GNU General Public License
+  ~ along with Watch Later.  If not, see <http://www.gnu.org/licenses/>.
+  -->
+
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.lambdasoup.tea_junit4">
+
+</manifest>
```
Comment (maxhille) [inline tea-junit4/src/main/AndroidManifest.xml:27]: newline

File: tea-junit4/src/main/java/com/lambdasoup/tea_junit4/TeaIdlingResource.kt [hunk 5]
```diff
@@ -20,12 +20,14 @@
  * along with Watch Later.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-package com.lambdasoup.tea
+package com.lambdasoup.tea_junit4
```
Comment (maxhille) [inline tea-junit4/src/main/java/com/lambdasoup/tea_junit4/TeaIdlingResource.kt:23]: (This is just a suggestion, no action required)

This package name is syntactically correct but against convention [0]. I think `com.lamdasoup.tea.junit4` and also nesting the gradle package would be good.

[0] https://docs.oracle.com/javase/tutorial/java/package/namingpkgs.html

**Review by maxhille — APPROVED** [review]

File: tea-junit4/.gitignore (hunk 1 above)
Comment (strooooke) [reply tea-junit4/.gitignore:1]: This answer seems quite out of date.

File: tea-junit4/src/main/java/com/lambdasoup/tea_junit4/TeaIdlingResource.kt (hunk 5 above)
Comment (strooooke) [reply tea-junit4/src/main/java/com/lambdasoup/tea_junit4/TeaIdlingResource.kt:23]: As discussed - took clues from how okhttp does it. No nesting, but proper package name.

**Review by maxhille — APPROVED** [review]

**Pull request merged**